packages-web/maps/src/components/GoogleMap.tsx

```diff
@@ -0,0 +1,194 @@
+        if (map.current) {
+            if (!bounds) {
+                bounds = bounds = new google.maps.LatLngBounds();
+            }
+            bounds.extend({
+                lat: marker.latitude,
+                lng: marker.longitude
+            });
+            const mapMarker = new google.maps.Marker({
+                position: {
+                    lat: marker.latitude,
+                    lng: marker.longitude
+                },
+                icon: marker.url
+                    ? {
+                          url: marker.url,
+                          scaledSize: new google.maps.Size(32, 32)
+                      }
+                    : "",
+                title: marker.title
+            });
+            if (marker.title) {
+                const infoContent = document.createElement("span");
+                infoContent.innerHTML = marker.title || "";
+                if (marker.onClick) {
+                    infoContent.style.cursor = "pointer";
+                    infoContent.onclick = marker.onClick;
+                }
+                const infoWindow = new google.maps.InfoWindow({
+                    content: infoContent
+                });
+                mapMarker.addListener("click", () => {
+                    infoWindow.open(map.current, mapMarker);
+                });
+            } else {
+                if (marker.onClick) {
+                    mapMarker.addListener("click", () => {
+                        marker.onClick!();
+                    });
+                } else {
+                    mapMarker.setClickable(false);
+                }
+            }
+            mapMarker.setMap(map.current);
+            return mapMarker;
+        }
+        return undefined;
+    };
+
+    const updateCamera = (): void => {
+        const { zoomLevel, autoZoom } = props;
+        setTimeout(() => {
+            if (bounds && map.current) {
+                try {
+                    if (!autoZoom) {
+                        map.current.setCenter(getCenter());
+                        map.current.setZoom(zoomLevel);
+                    } else {
+                        map.current.fitBounds(bounds);
+                    }
+                } catch (error) {
+                    setValidationMessage(`Invalid map bounds ${error.message}`);
+                }
+            }
+        }, 0);
+    };
+
+    const getCenter = (): google.maps.LatLngLiteral => {
+        return {
+            lat: markers.current?.[0]?.getPosition()?.lat() ?? defaultCenterLocation.lat,
+            lng: markers.current?.[0]?.getPosition()?.lng() ?? defaultCenterLocation.lng
+        };
+    };
+
+    const getMapStyles = (): google.maps.MapTypeStyle[] => {
+        if (props.mapStyles && props.mapStyles.trim()) {
+            try {
+                return JSON.parse(props.mapStyles);
+            } catch (error) {
+                setValidationMessage(`invalid Map styles, ${error.message}`);
```

Comment:
Maybe also good add some guidance in the doc about which structure do we expect for styles, so customers doesn't complain like :
`But my styles are correct ! your widget doesnt work.`

Reply:
Is in the description of the field (Studio side)